### ISSUE TITLE
How to handle a verse which is spread over two pages, with commentary and possibly apparatus in between?

### ISSUE THREAD

**Issue by ppasedach** [chchch/upama#14]
I am proofreading some [OCR-ed scan](https://github.com/sanskrit/raw_etexts/blob/master/kAvyam/mishram/kAvyamAlA_ocr/KavyamalaVol_22-HaravijayaOfRajanakaRatnakara1890.md) which I converted to IAST, and added TEI encoding. Many times you will get two pādas of a verse typeset on page n, then you get the commentary on the verses on that page, and then on pages n+1 you get the remaining verse half, after the page header.

![Selection_179](https://user-images.githubusercontent.com/11517856/130345187-86da9ae7-04b7-406f-9232-a5b1c760e484.png)

So far I was basically interested in collating the verses of the poem, so it was no problem to just comment out the commentary, page headers etc. 

```
<lg type="verse" xml:id="HV_26_20">
<l>vispaṣṭalakṣmaṇi saśīdhutaraṅgaratna-</l>
<l>pārītalapratimite śaśini priyāyāḥ |</l>
<!--
śobhā yasya || 14 || teṣāṃ niḥśvāsānāṃ saurabhasya yo'tiśayastena vāsanaṃ saṃskāramiva
kartumagre'tiṣṭhat | ivaśabdasya bhinnakramatve yuktiruktā || 15 || 16 ||17|| anyābhi|
dhā sapatnīnāma | rasacchaṭābhiriva dyutibhirāplāvitā pārī śuśubhe ||18|| āvarjitaṃ
dattam | udanaṃsīdunnanāma | 'yamaramanamātāṃ sakca' iti sagiṭī ||19|| 20 | candrasya

Page 334****************************************************************************************
328
kāvyamālā|
-->
<l>gaṇḍasthale ca vikacotpalakarṇapūra-</l>
<l>kalmāṣabhāsi bibhide na janasya dṛṣṭiḥ || 20 ||</l>
</lg>
```
But I want to soon start including the commentary also. My first idea would be to wrap the commentary to every verse in a `<p>`, and give them `xml:id`s through which they could be programmatically be associated to the verse they are on. As far as I can see, a `<p>` included in an `<lg>` appears in the apparatus. Which is something you may want every now and then, but not here obviously. 

Another problem is that `<p>`s are in Upama also represented as paragraphs, i.e. adding a line break after them, but here the edition renders the commentary on all verses on a page as a single paragraph. So maybe one would want to use `<div>`s of a certain type or something else within a single `<p>` for the whole typographical commentary block?

Yet another problem is that also the commentary on any verse can be spread over two pages, as for example the commentary on verse 14 here. 

One could of course split verses as well as commentary manually, like 
```
<lg xml:id="HV_26_20ab">
<l>
<l>
</lg>
<p>...</p>
<lg xml:id="HV_26_20cd">
<l>
<l>
</lg>
```
But that would then have to be done with every witness in order for the collation to work.

**Comment by ppasedach** [comment]
Ideally one would separate typographical structure from the logical structure, and have some way of specifying that a verse, or a piece of commentary is split up over two pages, such that Upama can then join its two halves before collating it with the other witnesses. Does that make sense?

**Comment by chchch** [comment]
I think this is described in TEI-speak as "textual streams":

https://tei-c.org/release/doc/tei-p5-doc/fr/html/ref-layout.html

In this case, the document has two "streams" per page. There's possibly a clunky solution to transcribing this by using `<floatingText>`, but I think it would be simpler and more readable to just, as you say, separate out the logical structure and use `<lg>` for the integral verses and `<p>` for the integral paragraphs:

`<lg xml:id="HV_26_20">`
`<l></l>`
`<l></l>`
`</lg>`

`<p>...</p>`

And treat the `<lg>` block and `<p>` block as different streams, so that the line numbering is separate.

**Comment by ppasedach** [comment]
I will need to rather soon find a good solution for this one. I do have a second manuscript worth transcribing at least for those cantos I critically edit, which has both verses and a commentary, with the verses in the middle of the page, and the commentary above and below it.

![jodhpur11123PA050059](https://user-images.githubusercontent.com/11517856/133447798-54395750-4c65-42b8-a44a-ab5e224403b2.JPG)


So do I understand "two streams" correctly that I would then just keep the two texts separate in my transcript, such that I'd, in theory, first have a transcript of all the verses, in which I plainly ignore the presence of the commentary in the manuscript, inserting a `<pb>` before `tarajaḥ`, make the next line `<lb n="2"/>`, and so on, and then only after the transcript of the verses of the whole manuscript have the transcript of the commentary, in which again I ignore the presence of the verses in between the upper and lower block of commentary? Upper and lower block of commentary could be provisionally separated by prefixing their numbers with u and l respectively. The `@n` attribute seems to be essentially a string, so that should not break things.

**Comment by chchch** [comment]
Yes I think that would work best. Actually what would be nice is that, for the bottom half of the commentary, we could have something like `<lb n="5" type="skip"/>` to indicate that that line skips over the mūla text in the middle. Or, we could even do something like `<cb type="vertical"/>`, since it's knd of like a new column.

I think I would actually put the commentary blocks after each mūla block that is being commented upon, but it doesn't really matter as long as the `xml:id`s match up.
